app/webfront/src/redux/recipes/operations.js

```diff
@@ -2,14 +2,26 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 import api from 'src/api/index.js';
 import { FETCH_RECIPES, ADD_RECIPE, DELETE_RECIPE } from './constants.js';
 
-export const fetchRecipes = createAsyncThunk(FETCH_RECIPES, async (_, { rejectWithValue }) => {
-    try {
-        const recipes = await api.getRecipes();
-        return recipes;
-    } catch (error) {
-        return rejectWithValue(error.message);
+export const fetchRecipes = createAsyncThunk(
+    FETCH_RECIPES,
+    async ({ page = 1, limit = 12, category = null, area = null, ingredients = [] }, { rejectWithValue }) => {
+        try {
+            const queryParams = new URLSearchParams();
+
+            if (page) queryParams.append('page', page);
+            if (limit) queryParams.append('limit', limit);
+            if (category && category !== 'all') queryParams.append('category', category);
+            if (area) queryParams.append('area', area);
+            if (ingredients && ingredients.length > 0) queryParams.append('ingredients', ingredients.join(','));
```

Comment:
Just a suggestion of simplification through spread operation:
   const queryParams = new URLSearchParams({
                ...(page && { page }),
                ...(limit && { limit }),
                ...(category && category !== 'all' && { category }),
                ...(area && { area }),
                ...(ingredients.length > 0 && { ingredients: ingredients.join(',') }),
            });

Reply:
No need to change, just for future)